core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -182,18 +180,25 @@ private[op] object FeatureDistribution {
    * @param values values to bin
    * @param sum summary info for feature (max and min)
    * @param bins number of bins to produce
-   * @param hasher hasing function to use for text
    * @return the bin information and the binned counts
    */
   // TODO avoid wrapping and unwrapping??
   private def histValues(
     values: ProcessedSeq,
     sum: Summary,
-    bins: Int,
-    hasher: HashingTF
+    bins: Int
   ): (Array[Double], Array[Double]) = {
     values match {
-      case Left(seq) => Array(sum.min, sum.max) -> hasher.transform(seq).toArray // TODO use summary info to pick hashes
+      case Left(seq) => {
+        val minBins = bins
+        val maxBins = MaxBins
+        val numBins = math.min(math.max(bins, sum.max/AvgBinValue), maxBins).toInt
+
+        val hasher: HashingTF = new HashingTF(numFeatures = numBins)
```

Comment:
do we have to create hasher every time or perhaps we can create it once?

Reply:
the hashing dimension can be different for different features, the minimum number would be `bins`
but we can have a shared one with numFeatures = bins, and use that for every case if there is not too many tokens; OR we can create a couple shared hashers with different scales, and choose one based on the scale of token numbers
What do you think? I was assuming creating a hasher for every feature will not be very resource consuming

Reply:
let see if we can reuse the hashing function without creating `HashingTF` everytime.

Reply:
and I think we can. See `HashingTF.transform` and `HashingTF` object